### PR TITLE
[TECH-521] Rename cvs to vcs

### PR DESCRIPTION
**Pull request by SamTheisens**
And add upgrade mechanism for `mpyl_config.yml`

Screenshot 2023-09-22 at 17 54 19

branch: feature/TECH-521-implement-remote-artifact-caching

----
### 📕 [TECH-491](https://vandebron.atlassian.net/browse/TECH-491) MPyL Improvements danielkoves@vandebron.nl 
Improve and extend our CI-CD capabilities

**DoD:** 

* MPL is marked as deprecated
* Most development run on MPyL

🏗️ Build [6](https://jenkins.k8s-serv-backend.vdbinfra.nl/job/MPyL%20Pipeline%20-%20Test/job/PR-253/6/display/redirect) ✅ Successful, started by _Sam Theisens_  
🚀 *[cloudfront-service](https://cloudfront-service-253.test.nl/swagger/index.html)*, *example-dagster-user-code*, *job*, *kong-sync*, *[nodeservice](https://nodeservice-253.test.nl/swagger/index.html)*, *[sbtservice](https://sbtservice-253.test.nl/swagger/index.html)*, *sparkJob*  

[TECH-491]: https://vandebron.atlassian.net/browse/TECH-491?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/mpyl/cli/commands/health/checks.py [hunk 1]
```diff
@@ -146,6 +171,25 @@ def __check_config(console, env_var, default, schema_path, name):
...
 
+        path, diff = check_upgrade_needed(path, upgraders)
+        pretty_diff = pretty_print(diff) if diff else ""
+        if pretty_diff == "":
+            console.check("Upgrade not necessary", success=True)
+        else:
+            console.check("Upgrade required", success=False)
+            console.print("Expected changes:")
+            console.print(pretty_diff)
+            if sys.stdout.isatty() and Confirm.ask("Upgrade now?"):
```
Comment (koeves) [inline src/mpyl/cli/commands/health/checks.py:182]: How is this going to work in the pipeline? Wouldn't the pipeline get stuck if upgrade is needed?

File: tests/test_resources/upgrades/mpyl_config_upgraded.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
+mpylVersion: 1.0.12
```
Comment (peterrutg) [inline tests/test_resources/upgrades/mpyl_config_upgraded.yml:1]: Are these the MPyL release versions or will we have a separate versioning for the config structure?

File: src/mpyl/cli/commands/health/checks.py (hunk 1 above)
Comment (SamTheisens) [reply src/mpyl/cli/commands/health/checks.py:182]: The `sys.stdout.isatty()` should prevent that from happening.

File: tests/test_resources/upgrades/mpyl_config_upgraded.yml (hunk 2 above)
Comment (SamTheisens) [reply tests/test_resources/upgrades/mpyl_config_upgraded.yml:1]: This supposed to follow the release semver scheme. With these intermediate versions it doesn't matter much what they are, but when for the multiple registries change we could target `1.3.0`

File: src/mpyl/cli/commands/health/checks.py [hunk 3]
```diff
@@ -53,24 +54,32 @@ def perform_health_checks(bare_console: Console, is_ci: bool = False):
     __check_version(console)
 
     console.title("MPyL configuration")
-    __check_config(
+    if config_path := __validate_config_path(
```
Comment (koeves) [inline src/mpyl/cli/commands/health/checks.py:59]: Huh I didn't know we have the walrus operator in python

**Review by koeves — APPROVED** [review]
Looks good!

File: src/mpyl/cli/commands/health/checks.py (hunk 3 above)
Comment (SamTheisens) [reply src/mpyl/cli/commands/health/checks.py:59]: Yes, I like it, reminds me of pascal 😄 . And (way too) hard fought https://medium.com/pythonistas/the-story-of-guido-van-rossum-and-his-walrus-operator-quick-blog-a0481af61bd8

**Pull request merged**